app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.html

```diff
@@ -59,63 +59,128 @@
       </div>
     </div>
 
-    <div class="form-group">
-      <div class="col-md-3 sm-label-right">
-        Git Repository URL <help-field class="help-field-absolute" key="appengine.serverGroup.repositoryUrl"></help-field>
+    <div class="form-group row">
+      <label class="col-md-3 sm-label-right">Source Repository Type</label>
+      <div class="col-md-9">
+        <select class="input-sm"
+                ng-options="option for option in basicSettingsCtrl.sourceRepositoryTypes"
+                ng-model="command.sourceRepositoryType">
+          <option value="">Select...</option>
+        </select>
       </div>
-      <div class="col-md-7">
-        <input type="text"
-               required
-               class="form-control input-sm"
-               name="gitRepo"
-               ng-model="command.repositoryUrl"/></div>
     </div>
 
-    <div class="form-group">
-      <div class="col-md-3 sm-label-right">
-        Git Credential Type <help-field class="help-field-absolute" key="appengine.serverGroup.gitCredentialType"></help-field>
+    <div ng-if="basicSettingsCtrl.isArtifactSource()">
+      <div class="form-group">
+        <label class=" col-md-3 sm-label-right">Expected Artifact</label>
+        <div class="col-md-7">
+          <ui-select ng-if="command.backingData.expectedArtifacts.length"
```

Comment:
Having an empty dropdown here is fine - or you could have a message that says something like `You have not configured any expected artifacts` - but blank space is confusing.